cocos/base/CCScheduler.cpp

```diff
@@ -547,10 +548,16 @@ void Scheduler::removeUpdateFromHash(struct _listEntry *entry)
     HASH_FIND_PTR(_hashForUpdates, &entry->target, element);
     if (element)
     {
-        // list entry
-        DL_DELETE(*element->list, element->entry);
-        CC_SAFE_DELETE(element->entry);
-
+        if (_updateHashLocked)
+        {
+            element->entry->markedForDeletion = true;
+        }
+        else
+        {
+            // list entry
+            DL_DELETE(*element->list, element->entry);
+            CC_SAFE_DELETE(element->entry);
+        }
```

Comment:
removeUpdateFromHash should not consider `_updateHashLocked `, it just remove from hash, or it will make things complex as you added the codes following.